gui/src/main.rs

```diff
@@ -242,6 +239,41 @@ impl App for Gui {
                 }
                 ActiveWindow::HISTORY => {
                     ui.label("History");
+                    let history_items_clone = Rc::clone(&self.request_history_items);
+                    let history_items = history_items_clone.borrow();
+                    if let Some(item_vec) = &*history_items {
+                        for item in item_vec {
+                            if ui.selectable_value(
+                                &mut self.selected_history_item,
+                                Rc::new(RefCell::from(Some(item.clone()))),
+                                format!("{}", item.id)
+                            ).clicked() {
```

Comment:
For each request history item that is clicked, we look up the full details from the two above HashMaps and then update the appropriate ui bits.